### PR TITLE
Update Librem boards to coreboot 4.17

### PR DESCRIPTION
**Pull request by danielp96**
Adds coreboot 4.17 support to heads.
coreboot 4.17 solves CVE-2022-29264 which started to affect coreboot 4.13 when "CONFIG_X86_SMM_LOADER_VERSION2" was optionally activated (not the default), and where default SMM_LOADER2 became the default under in 4.14+.

The corresponding mailing thread goes into this in more extent: https://mail.coreboot.org/hyperkitty/list/coreboot@coreboot.org/thread/PAT6C2PW7G35I5IVJOBVYRRNZMU4IGUX/#DQHCWQO65KPAOJ2HTE5FY64ZHTO64ECD

### CODE REVIEW

**Comment by tlaurion** [comment]
@danielp96 Hello there!

Can I replace MrChromebox with danielp96 handle  or all Librem devices under https://github.com/osresearch/heads/issues/692?

I also felt free to edit OP to give a little bit more detail about the issue. 
Feel free to modify accordingly with what you feel more comfortable with.

LGTM!  

Where CircleCI is changing 4.15 cache with 4.17 here, so no problem with CirrcleCI config change. 
Next time, it might need to be modified accordingly so that the caching is effective: https://app.circleci.com/pipelines/github/osresearch/heads/439/workflows/f6c61946-e001-4bbe-b89e-55942864a9ce

**Comment by tlaurion** [comment]
@danielp96 ping for answers on above questions.

This PR also needs rebasing on master.

**Comment by JonathonHall-Purism** [comment]
Hey @tlaurion :wave: , I'm catching up from my absence last week, thanks for checking this out, and thanks @danielp96 for getting it together.

>Can I replace MrChromebox with danielp96 handle or all Librem devices under https://github.com/osresearch/heads/issues/692?

Please replace MrChromebox with me for all Librem devices, I may delegate to Daniel but I'm the primary firmware engineer for Purism.

It looks like we can remove coreboot 4.15, no other devices depend on it and it is affected by this CVE.  @danielp96 could you take care of that?  Please rebase on upstream master as well.

**Comment by danielp96** [comment]
Rebased and removed support for coreboot 4.15

File: modules/coreboot [hunk 1]
```diff
@@ -29,16 +29,17 @@ else ifeq "$(CONFIG_COREBOOT_VERSION)" "4.13"
...
 	coreboot_depends := $(if $(CONFIG_PURISM_BLOBS), purism-blobs)
 	EXTRA_FLAGS := -fdebug-prefix-map=$(pwd)=heads -gno-record-gcc-switches -Wno-error=packed-not-aligned -Wno-error=address-of-packed-member
 else ifeq "$(CONFIG_COREBOOT_VERSION)" "talos_2"
-	coreboot_version = git
-	coreboot_commit_hash = 2207bbcccba31ad89cf21607b0d8d05d8dc47c03
-	coreboot_repo := https://github.com/Dasharo/coreboot
+    coreboot_version = git
+    coreboot_commit_hash = 2207bbcccba31ad89cf21607b0d8d05d8dc47c03
+    coreboot_repo := https://github.com/Dasharo/coreboot
+
```
Comment (JonathonHall-Purism) [inline modules/coreboot:42]: @danielp96 Your editor might have tweaked the indentation on these lines

**Comment by JonathonHall-Purism** [comment]
@danielp96 Could you also remove the patches/coreboot-4.15 directory?

File: modules/coreboot (hunk 1 above)
Comment (danielp96) [reply modules/coreboot:42]: Didn't notice, gonna fix.

**Comment by danielp96** [comment]
Removed patches/coreboot-4.15

**Comment by tlaurion** [comment]
I kicked the build which was stalled.

**Comment by tlaurion** [comment]
@danielp96 : Please make sure your commits are signed.

**Comment by danielp96** [comment]
Commits verified and circleci running under my instance.

**Comment by tlaurion** [comment]
> Please replace MrChromebox with me for all Librem devices, I may delegate to Daniel but I'm the primary firmware engineer for Purism.

@JonathonHall-Purism done.
Successfully built, I consider that this coreboot bump was tested functional. Merging!

**Pull request merged**